nomad/state/state_store.go

```diff
@@ -1741,8 +1741,11 @@ func (s *StateStore) CSIVolumeClaim(index uint64, id string, alloc *structs.Allo
 		return fmt.Errorf("volume row conversion error")
 	}
 
-	volume := orig.Copy()
-
+	ws := memdb.NewWatchSet()
+	volume, err := s.CSIVolumeDenormalizePlugins(ws, orig.Copy())
```

Comment:
This uncovered an interesting bug: the volume claim step was (incorrectly) relying on our updates of the struct, which would mean this would probably work in single-node testing and then fail once we tested with a cluster.